### PR TITLE
Workaround horrible google recurring rules for contacts birthdays. Other recurring event bugs.

### PR DESCRIPTION
**Pull request by lightsofapollo**
Also cleanup performance test and adds benchmark for the previous version of ical.js (I regressed a bit to make this change possible but its not too horrible).

### CODE REVIEW

File: lib/ical/design.js [hunk 1]
```diff
@@ -571,7 +581,13 @@ ICAL.design = (function() {
       "rdate": {
         defaultType: "date-time",
         allowedTypes: ["date-time", "date", "period"],
-        multiValue: ','
+        multiValue: ',',
+        detectType: function(string) {
```
Comment (lightsofapollo) [inline lib/ical/design.js:585]: I thought that the VALUE= must be specified rather then just using whatever type you felt like in RDATE but google does not agree in all cases.

File: lib/ical/event.js [hunk 2]
```diff
@@ -48,7 +59,7 @@ ICAL.Event = (function() {
         obj = new ICAL.Event(obj);
       }
 
-      if (obj.uid !== this.uid) {
+      if (this.strictExceptions && obj.uid !== this.uid) {
```
Comment (lightsofapollo) [inline lib/ical/event.js:62]: Another option was to do a `object.uid.indexOf(this.uid) !== -1` and get rid of the strict option but this was a bit faster and you will likely want strict on if parsing a large ICAL file rather then small caldav responses.

File: lib/ical/recur_expansion.js [hunk 3]
```diff
@@ -289,16 +289,16 @@ ICAL.RecurExpansion = (function() {
...
 
       for (; i < len; i++) {
-        prop = props[i].getFirstValue();
-
-        idx = ICAL.helpers.binsearchInsert(
-          result,
-          prop,
-          compareTime
-        );
+        props[i].getValues().forEach(function(prop) {
```
Comment (lightsofapollo) [inline lib/ical/recur_expansion.js:292]: Real bug I was not expecting RDATE/EXDATE to be multivalued. Now we correctly handle both multiple properties and multiple values per property.

**Comment by KevinGrandon** [comment]
I took a look, r+=me.

**Comment by lightsofapollo** [comment]
I had Kevin Grandon review this for me, we can revisit issues later if there are any.

**Pull request merged**